lib/TPP/Transforms/TileConsumerAndFuseProducers.cpp

```diff
@@ -574,7 +587,7 @@ static Operation *getLastFusableEltWiseConsumer(
   Value linalgOpRes = linalgOp->getResult(0);
   // If we allow use, we may end up doing recomputation. Unclear if it is
   // profitablem thus disallow for now.
-  if (!linalgOpRes.hasOneUse())
+  if (!hasActualOneUse(linalgOpRes))
```

Comment:
Tile-and-fuse is performed at the tensor level where tensor are immutable and there's no concept of overwriting buffer values. If the linalg op had memref semantics, it would have no results.

I'm not sure how `hasOneUse` could be insufficient. Do you have any specific example?

Reply:
You are right to raise this question and it basically relates to why with such a tensor semantic we have this DPS representation.
However, given that one can have "linalg.add ins(%A, %B) outs(%A)", hasOneUse() API, which is based on SSA
representation, will count it as >1 use while in fact the user is only one.

I stumbled upon this test case "matmul_sequence_fusion_with_eltwise" in tile-and-fuse.mlir. This was not getting fused for the named op equivalent test case.

Reply:
Thanks for the explanation! Now I see clearly what you're trying to solve.
You're right, this is a case we should be able to handle.